libmscore/textbase.cpp

```diff
@@ -2476,12 +2476,48 @@ int TextBase::getPropertyFlagsIdx(Pid id) const
       return -1;
       }
 
+//---------------------------------------------------------
+//   offsetTid
+//---------------------------------------------------------
+
+Sid offsetTid(Tid tid, bool placeAbove)
```

Comment:
To sum up what I have said above, this could probably be moved to `TextBase` class as a static member as it deals with text-related issues only...

Reply:
If `offsetTid()` is to become a method of `class TextBase`, why make it static? Making it an object method would eliminate the need to pass `tid` and `placeAbove` as parameters, since these are properties of TextBase elements.

Reply:
Also, wouldn't it make more sense to name this function `offsetSid`, since it returns an `Sid` and not a `Tid`? That was the name I originally gave it, and I do not understand why it was changed.

Reply:
Weird, I would have sworn I just copied/pasted your code.  But I guess I just took the body of the function and retyped the declaration, and missed it by *that* much.  I have to admit, I got a certain kick out of having written in the original PR - in all seriousness - "we need to check Tid before mapping Pid to Sid."  But yeah, it's easy to lose track of what's what.

Anyhow, the reason to make this a static member - if I'm not confusing myself further - is as per the comment from @dmitrio95 : the function really is more about Tid than element type. 
 This distinction was more or less the reservation I had with your original code.  To be honest, my head is spinning a little.

Anyhow, I'm going to push an update shortly, I'll try to make sure it makes sense.

Reply:
I do not see how that is a reason to not make it an object method. I am not talking about overriding virtual functions; I am talking about having the method use the object's own `tid()` and `placeAbove()` values instead of having to pass them as parameters.